src/communication/network_interface_sr.c

```diff
@@ -10190,3 +10191,114 @@ ssession_stop_attached_threads (void *session)
 {
   session_stop_attached_threads (session);
 }
+
+void
+smethod_invoke_fold_constants (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
+{
+  packing_unpacker unpacker (request, (size_t) reqlen);
+
+  method_sig_list sig_list;
+  sig_list.unpack (unpacker);
+
+  int arg_cnt = 0;
+  unpacker.unpack_int (arg_cnt);
+
+  /* *INDENT-OFF* */
+  std::vector<DB_VALUE> args (arg_cnt);
+  /* *INDENT-ON* */
+  for (int i = 0; i < arg_cnt; i++)
+    {
+      unpacker.unpack_db_value (args[i]);
+    }
+
+  /* *INDENT-OFF* */
+  cubmethod::method_invoke_group method_group (&sig_list);
+  /* *INDENT-ON* */
+
+  method_group.begin (thread_p);
+
+  int error_code = method_group.prepare (args);
+  if (error_code != NO_ERROR)
+    {
+      return_error_to_client (thread_p, rid);
```

Comment:
return_error_to_client returns TRAN_STATE, do not need to process the return value?

Reply:
ok, I'm going to cast with `(void)`